app/src/ui/lib/tooltipped-commit-sha.tsx

```diff
@@ -1,51 +1,68 @@
 import { clipboard } from 'electron'
 import React from 'react'
+import { Commit, shortenSHA } from '../../models/commit'
+import { Ref } from './ref'
 import { TooltipDirection } from './tooltip'
 import { TooltippedContent } from './tooltipped-content'
 
 interface ITooltippedCommitSHAProps {
   readonly className?: string
-  /**
-   * Short SHA to display in the tooltipped element, or the tooltipped element
-   * itself.
-   */
-  readonly shortSHA: string | JSX.Element
-
-  /** Long SHA to display within the tooltip. */
-  readonly longSHA: string
+  /** Commit or long SHA of a commit to render in the component. */
+  readonly commit: string | Commit
+
+  /** Whether or not render the commit as a Ref component. Default: false */
+  readonly asRef?: boolean
 }
 
 export class TooltippedCommitSHA extends React.Component<
   ITooltippedCommitSHAProps,
   {}
 > {
+  private get shortSHA() {
+    const { commit } = this.props
+    return typeof commit === 'string' ? shortenSHA(commit) : commit.shortSha
+  }
+
+  private get longSHA() {
+    const { commit } = this.props
+    return typeof commit === 'string' ? commit : commit.sha
+  }
+
   public render() {
-    const { className, shortSHA } = this.props
+    const { className } = this.props
 
     return (
       <TooltippedContent
         className={className}
-        tooltip={this.renderShaTooltip()}
+        tooltip={this.renderSHATooltip()}
         tooltipClassName="sha-hint"
         interactive={true}
         direction={TooltipDirection.SOUTH}
       >
-        {shortSHA}
+        {this.renderShortSHA()}
       </TooltippedContent>
     )
   }
 
-  private renderShaTooltip() {
+  private renderShortSHA() {
+    return this.props.asRef === true ? (
+      <Ref>{this.shortSHA}</Ref>
+    ) : (
+      <>{this.shortSHA}</>
```

Comment:
```suggestion
      this.shortSHA
```

The `<>` are not needed. :)

Reply:
Oh! That makes sense 😂 Thanks, fixed in [a571196](https://github.com/desktop/desktop/pull/15220/commits/a571196942913804345ffb5a8825d4317c710c2e)!